### ISSUE TITLE
CSV Files Not Found (CORS error)

### ISSUE THREAD

**Issue by mDisna** [NewPath-Consulting/ez-wildapricot-webdesigner#168]
There appears to be a change to how WildApricot is storing files which is likely affecting all EZ Designer users. Links to uploaded files seem to be redirecting to a CDN, causing a 404 error since the CDN is not part of the CORS.

This is likely an issue that needs to be address on WA's side, but until they do some functionality may not be working correctly on the Designer.

**Comment by mDisna** [comment]
https://gethelp.wildapricot.com/en/articles/2847-version-8-7-0-release

**Comment by mDisna** [comment]
Unfortunately, adding `cdn.wildapricot.com` to the JavaScript whitelist does not fix the issue.

Error message: `No 'Access-Control-Allow-Origin' header is present on the requested resource.`

**Comment by mDisna** [comment]
@asirota  Looks like the error is back.

**On version 2.x, the CSV won't load and on 3.0 the JSON file won't load**
```
Access to XMLHttpRequest at 'https://cdn.wildapricot.com/...' (redirected from 'https://newpathconsulting.wildapricot.org/resources/Theme/WildApricotTextManager/config.csv') from origin 'https://newpathconsulting.wildapricot.org' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.
```

**On 3.0, there is also an error when trying to get a screenshot of the page now, needed for the palette ([Stackoverflow](https://stackoverflow.com/questions/22097747/how-to-fix-getimagedata-error-the-canvas-has-been-tainted-by-cross-origin-data))**
```
Error capturing the webpage: DOMException: Failed to execute 'getImageData' on 'CanvasRenderingContext2D': The canvas has been tainted by cross-origin data.
    at https://ezdesigner.wildapricot.org/resources/Theme/WildApricotTextManager/wildapricot-textmanager.js
```

**Comment by asirota** [comment]
Can you open a ticket with their support?

**Comment by mDisna** [comment]
The issue loading files seems to have been resolved. The issue with the palette function remains. I have opened support ticket #1620485, but will remove/disable the palette function in 3.0 for the time being.

**Comment by mDisna** [comment]
The issue loading the csv in 2.x has been resolved on WildApricot's end, however the issue still remains for the palette colours on 3.0.

**Comment by asirota** [comment]
That's good news! Glad we didn't have to do anything on our need. The palette issue probably will still be something we have to fix
On our end or just abandon it altogether.

**Comment by mDisna** [comment]
The palette function has been rewritten to get around the error message.
